client/allocrunnerv2/taskrunner/task_runner.go

```diff
@@ -287,42 +289,48 @@ MAIN:
 			tr.logger.Error("poststart failed", "error", err)
 		}
 
-		// Grab the handle
-		handle = tr.getDriverHandle()
-
-		select {
-		case waitRes := <-handle.WaitCh():
-			// Clear the handle
-			tr.clearDriverHandle()
+		// Grab the result proxy and wait for task to exit
+		{
```

Comment:
Why the new scope?

Reply:
Without scoping `result` you have to declare it outside the loop to avoid...
```
goto RESTART jumps over declaration of _, result 
```
...errors.

It's a bit ugly, but it's localized ugliness vs declaring a short-lived variable (result) for the life of the entire func.